AWSPinpoint/AWSPinpointEventRecorder.m

```diff
@@ -1149,4 +1268,20 @@ - (AWSPinpointTargetingPutEventsRequest*) putEventsRequestForEvents:(NSDictionar
     return putEventRequest;
 }
 
++ (NSMutableDictionary *)getMutableDictionaryFromResultSet:(AWSFMResultSet *)rs
+                                             forColumnName:(NSString *)columnName
+                                                     error:(NSError *__autoreleasing *)error {
+    NSSet *allowableClasses = [[NSSet alloc] initWithObjects:[NSMutableString class],
+                               [NSDictionary class],
+                               nil];
+    NSDictionary *immutableDict = [AWSNSCodingUtilities versionSafeUnarchivedObjectOfClasses:allowableClasses
+                                                                                    fromData:[rs dataForColumn:columnName]
+                                                                                       error:error];
+    if (*error) {
+        return nil;
```

Comment:
Log this error?

Reply:
Good callout, will do

Reply:
As discussed offline, I will add this in a subsequent PR